pages/videos.vue

```diff
@@ -34,17 +32,31 @@
       }
     },
     components: {
+      VideoGrid,
       VideoTableWithSearch
     },
     computed: {
       ...mapState({
         allVideos: state => state.videos.videos,
       }),
+      ...mapGetters({
+        getCourse: 'courses/get'
+      }),
       isMobile() {
         return this.$vuetify.breakpoint.xsOnly
       },
-      publishedVideos(){
-        return this.allVideos.filter(v => v.published_at < Date.now())
+      videoList () {
+        return this.allVideos.filter(v => v.published_at < Date.now()).map((v)=>{
+          let course = this.getCourse(v.course_id)
+          let courseName = course && course.name
+          courseName = courseName && courseName.length > 33  ? `${courseName.slice(0, 30)}...` : courseName
+          return {
+            ...v,
+            sortable_published_at: v.published_at && v.published_at.toISOString(),
+            courseName,
+            courseVideoNumber: course.video_ids.findIndex( i => parseInt(i) === v.id) + 1
```

Comment:
Where the course happened to be place in the array is not the correct formula for this.

For example, you click this, where it says "Video 1"

![Screen Shot 2020-05-04 at 4 55 08 PM](https://user-images.githubusercontent.com/839123/81024368-1166b080-8e28-11ea-84c7-40ae4bb5542e.png)

and you end up here, with the 3rd video

![Screen Shot 2020-05-04 at 4 55 14 PM](https://user-images.githubusercontent.com/839123/81024383-1fb4cc80-8e28-11ea-9fe2-445a3e19266c.png)

Reply:
Here's the relevant line in `/watch/_id`

```
sortedVideos() { return _.sortBy(this.course.videos, v => Number(v.order)) },
```